src/components/SearchView.tsx

```diff
@@ -32,6 +32,10 @@ export class SearchView extends React.Component<SearchViewProps, SearchViewState
 
     searchResultListItems: any;
 
+    //To hold the categories which will be displayed in the search bar.
+    //This will include both the selected and the unselected items.
+    displayedCategories: string[];
```

Comment:
Some suggestions:

```js
    // This list represents the actual categories beind displayed on the search bar 
    // at the time search is performed. It will be a subset of 'props.categories', 
    // that are most relevant to the current search results.
```